README.md

```diff
@@ -2,24 +2,198 @@
+
+#### Initial discussion
+
+The process of amending the PMIx Standard begins with the opening of a **GitHub Issue** describing the proposed change. The Issue template used and the degree of detail provided in the issue should be commensurate with the magnitude of the proposed modification - e.g., a spelling correction would use the "Questions & Clarification" template and require very little justification, while a new API or deprecation of an existing API might use the "Use-Case" template and reasonably be expected to be accompanied by a more detailed justification. Use of the template corresponding to the type of modification being proposed is helpful to expedient processing of the proposal - if in doubt as to the one to use, please ask.
+
+The Issue provides a discussion forum for the contribution without necessarily specifying precise changes to the PMIx Standard document. Once an Issue is ready for discussion, it is helpful, though not required, to send a message to the PMIx Forum mailing list to raise awareness. Note that an Issue may not result in an actual change to the Standard - e.g., it might serve simply as a focal point for community discussion regarding a potential use-case or problem arising from a use-case.
+
+If a change to the PMIx Standard is included in the issue (or results from the discussion in the issue), then a Pull Request must be created (see below). Note that multiple Pull Requests may reference a single Issue as authors consider alternative approaches to addressing the Issue. The Issue will be closed once the associated PRs have been resolved (either through by being accepted, withdrawn, or rejected) or once the participants are satisfied with the conversation.
+
+#### Proposed modification
+When a specific change to the PMIx Standard has been identified a PR must be opened with a reference made to the corresponding Issue(s). When the PR is opened it should be placed in a [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) state until the following conditions are met:
+
+* The contents of the PR are ready for review and broad discussion with the PMIx community. Note that discussion over the PR is encouraged at all times even while in draft form.
+
+* An open-source implementation of the proposal is available and a reference to the implementation included in the PR. Note that a complete implementation is not necessarily required to meet this condition - e.g., pseudocode detailing the execution path supporting the proposed modification can be provided.More complete prototype implementations may be requested by the PMIx community when proposals involving significant changes in behavior are made.
```

Comment:
Is the idea of allowing psuedocode to better support closed-source implementations that do not want to release their full implementation of a feature and instead provide a rough sketch?

Minor nit: missing a space after a full stop
```suggestion
* An open-source implementation of the proposal is available and a reference to the implementation included in the PR. Note that a complete implementation is not necessarily required to meet this condition - e.g., pseudocode detailing the execution path supporting the proposed modification can be provided. More complete prototype implementations may be requested by the PMIx community when proposals involving significant changes in behavior are made.
```

Reply:
The pseudocode idea was in fact driven by the closed-source concern (not just an entire implementation - they might have a proprietary plugin that implements the change), but it also seemed to me that we don't always require an in-depth coding example right away (e.g., for some fairly trivial modification). So someone could provide us initially with pseudocode and the Members can decide if a more detailed implementation needs to be seen prior to provisional approval.

I don't have strong feelings on it - any thoughts?

Reply:
If I were reviewing an extensions for inclusion under the "Provisional" class, I believe that I would be OK with psuedo-code.  For moving to the "Stable" class, I would feel very strongly about requiring an open-source implementation be referenced, so that I could take the implementation into account (how difficult it appears to correctly implement, does the documented semantics align with the implemented semantics, etc).  I'm not sure if others agree enough that we want to document that as a hard requirement, but that is at least my thoughts.

Reply:
Yeah, I agree. However, I'm not sure how to handle proprietary implementations in that scenario. Probably something we should discuss in tomorrow's WG meeting.